var/spack/repos/builtin/packages/ferret/package.py

```diff
@@ -0,0 +1,103 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+import os
+
+
+class Ferret(Package):
+    """Ferret is an interactive computer visualization and analysis environment
+       designed to meet the needs of oceanographers and meteorologists
+       analyzing large and complex gridded data sets."""
+    homepage = "http://ferret.noaa.gov/Ferret/"
+    url      = "ftp://ftp.pmel.noaa.gov/ferret/pub/source/fer_source.tar.gz"
+
+    version('6.96', '51722027c864369f41bab5751dfff8cc',
+            url="ftp://ftp.pmel.noaa.gov/ferret/pub/source/fer_source.tar.gz")
+
+    depends_on("hdf5~mpi~fortran")
+    depends_on("netcdf~mpi")
```

Comment:
Does it require netcdf without MPI, does it just not *need* MPI?  If you put the hard constraint, this will force Spack to always build with a `netcdf~mpi`, even if `netcdf+mpi` is available.